--- a/src/gd.h	2024-01-07 16:51:43.749223000 +0100
+++ b/src/gd.h	2024-01-07 16:52:34.162291600 +0100
@@ -45,7 +45,7 @@
    the gd sources in a project. */
 
 /* http://gcc.gnu.org/wiki/Visibility */
-#if defined(_WIN32) || defined(CYGWIN) || defined(_WIN32_WCE)
+#if 0 // Disable DLL annotations when building statically. Needed for embedding under MinGW.
 # ifdef BGDWIN32
 #  ifdef NONDLL
 #   define BGD_EXPORT_DATA_PROT
